### PR TITLE
SONAR-6761 Drop incremental mode

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java [hunk 1]
```diff
@@ -454,6 +454,7 @@
 
   /**
    * @since 4.0
+   * @deprecated since 5.2
    */
   String ANALYSIS_MODE_INCREMENTAL = "incremental";
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: INFO Do not forget to remove this deprecated code someday. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
MAJOR Add the missing @Deprecated annotation. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)

**Comment by SonarTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 2 major
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Change this instance-reference to a static reference.](https://github.com/SonarSource/sonarqube/blob/ec8fb72d2767edc9d53745307eaf8f2cde98f12c/sonar-core/src/main/java/org/sonar/core/util/DefaultHttpDownloader.java#L136) [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2209)

File: sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java (hunk 1 above)
Comment (henryju) [inline sonar-plugin-api/src/main/java/org/sonar/api/CoreProperties.java]: See issue: missing @Deprecated

**Comment by henryju** [comment]
LGTM

**Pull request closed without merging**